content/posts/2022-12-24-react-native-redux-toolkit/index.md

```diff
@@ -0,0 +1,503 @@
+                    dispatch(deposit(10));
+                }}
+            />
+        </View>
+        <View style={{ marginTop: 20 }}>
+            <Button
+                title="Withdraw"
+                onPress={() => {
+                    // We dispatch the withdraw action to the store with payload 10
+                    dispatch(withdraw(10));
+                }}
+            />
+        </View>
+    </View>
+  );
+}
+```
+
+Awesome! Now we can dispatch actions to the store. Let's see how the balance state is updated when we dispatch the actions.
+
+### Update the Balance State
+
+Go to the app and click on the `Deposit` button. You will see that the balance state is updated to 10$. Click on the `Withdraw` button. You will see that the balance state is updated to 0$.
+
+When we dispatch the `deposit` action, the `deposit` reducer function is called with payload value as 10. The `deposit` reducer function updates the `balance` state by adding the payload to the current balance. Similarly, when we dispatch the `withdraw` action, the `withdraw` reducer function is called. The `withdraw` reducer function updates the `balance` state by subtracting the payload from the current balance.
+
+Every time the `balance` state is updated, the `TabOneScreen` component is re-rendered. This is because we are using the `useSelector` hook to access the `balance` state. The `useSelector` hook is called every time the `balance` state is updated. The `useSelector` hook returns the updated `balance` state and the `TabOneScreen` component is re-rendered with the updated `balance` state.
+
+To check if our `balance` state is available in other components, go to `TabTwoScreen` component and add the following code:
+
+```tsx
+import { useSelector } from 'react-redux';
+
+export default function TabTwoScreen({}: RootTabScreenProps<'TabTwo'>) {
+  // We use the useSelector hook to access the balance state from the store
+  const balance = useSelector((state) => state.balance.value);
+
+  return (
+    <View style={styles.container}>
+      <Text style={styles.title}>Tab Two</Text>
+      <View style={{ marginTop: 20 }}>
+        <Text style={{ fontSize: 20 }}>Current Balance: {balance}$</Text>
+      </View>
+    </View>
+  );
+}
+```
+
+Go to the app and click on the `Deposit` button. You will see that the balance state is updated to 10$ in both the `TabOneScreen` and `TabTwoScreen` components. Click on the `Withdraw` button. You will see that the balance state is updated to 0$ in both the `TabOneScreen` and `TabTwoScreen` components.
+
+## Bonus: Rect Native Debugger with Redux Extension
+
+We can use the [React Native Debugger](https://github.com/jhen0409/react-native-debugger) to debug our React Native app. This awesome debugger already includes [Redux DevTools Extension](https://github.com/zalmoxisus/redux-devtools-extension) to debug our Redux store. The advantages of redux devtools are:
```

Comment:
I see you link here to the tools you mention, but did not link above. I wanted to mention it, but I didn't know what's best for us, because if we link to the docs, people can leave the website. 

On the other hand, this is a good info for them to maybe open it up in a new tab and see more details afterwards. 

So, probably I would recommend linking (at least the first time) when you mention some external tools, libraries, etc.

Reply:
I did not fully understand. For which ones did you mean that I did not give the links?

Reply:
For example Redux, ReduxToolkit